Mathlib/Algebra/Order/Star/Basic.lean

```diff
@@ -391,3 +392,11 @@ instance Nat.instStarOrderedRing : StarOrderedRing ℕ where
         (AddSubmonoid.closure_mono <| singleton_subset_iff.2 <| mem_range.2 ⟨1, one_mul _⟩)
         Nat.addSubmonoid_closure_one
     simp [this, le_iff_exists_add]
+
+/-- a star projection is non-negative in a star-ordered ring -/
+theorem IsStarProjection.nonneg {M : Type*} [NonUnitalSemiring M] [PartialOrder M]
+    [StarRing M] [StarOrderedRing M] {p : M}
+    (hp : IsStarProjection p) : 0 ≤ p := by
```

Comment:
Can you also add the lemmas `0 ≤ 1 - p`, `p ≤ 1` and `p ∈ Set.Icc 0 1`, for convenience?